### PR TITLE
Lubien 012

### PR DESCRIPTION
**Pull request by lubien**
Intel® Core™ i3-5005U CPU @ 2.00GHz × 4 
8,44s

### CODE REVIEW

**Comment by lubien** [comment]
Approved.

**Comment by ryukinix** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/DestructHub/ProjectEuler/pull-request/60/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=DestructHub/ProjectEuler)

File: Problem012/Elixir/solution_1.exs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Author: lubien
+
+defmodule Euler12 do
+  def solve do
+    Stream.iterate(1, &(&1 + 1))
+      |> Stream.scan(&(&1 + &2))
+      |> Stream.drop_while(fn i -> count_divisors(i) < 500 end)
+      |> Enum.take(1)
+      |> hd
+  end
```
Comment (leorog) [inline Problem012/Elixir/solution_1.exs]: nice piece of code! :dancer:

Comment (lubien) [inline Problem012/Elixir/solution_1.exs]: @leorog  there's a trivia about this method
I was wondering how to make triangular numbers when I saw this in [Elixir Stream documentation](http://elixir-lang.org/docs/stable/elixir/Stream.html#scan/2):
```exs
iex> stream = Stream.scan(1..5, 0, &(&1 + &2))
iex> Enum.to_list(stream)
[1, 3, 6, 10, 15]
```

**Pull request merged**